falied to create azurerm_security_center_auto_provisioning

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.7.5

### AzureRM Provider Version

3.97.1

### Affected Resource(s)/Data Source(s)

azurerm_security_center_auto_provisioning

### Terraform Configuration Files

```hcl
resource "azurerm_security_center_auto_provisioning" "example" {
  auto_provision = "On"
}
```


### Debug Output/Panic Output

```shell
│ Error: updating auto-provisioning setting for Auto Provisioning Setting: (Name "default"): security.AutoProvisioningSettingsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="Deprecated" Message="Log Analytics auto provisioning is deprecated and can no longer be enabled."
2460│
2461│   with module.sece.azurerm_security_center_auto_provisioning.seccen,
2462│   on sece/main.tf line 10, in resource "azurerm_security_center_auto_provisioning" "seccen":
2463│   10: resource "azurerm_security_center_auto_provisioning" "seccen" {
2464│
2465╵
2466Error: Terraform exited with code 1.
2467Error: Process completed with exit code 1.
```


### Expected Behaviour

the resource should be created without any error

### Actual Behaviour

Can't create the resource because I think internally the provider tries to enable log analytics agent autoprovisioning which  will be deprecated by November 2024 [info](https://learn.microsoft.com/en-us/azure/defender-for-cloud/prepare-deprecation-log-analytics-mma-agent)

### Steps to Reproduce

1. user azurerm version 3.97.1
2. terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/securitycenter/registration.go
+++ b/internal/services/securitycenter/registration.go
@@ -4,6 +4,7 @@
 package securitycenter
 
 import (
+	"github.com/hashicorp/terraform-provider-azurerm/internal/features"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
 )
@@ -49,9 +50,12 @@ func (r Registration) SupportedResources() map[string]*pluginsdk.Resource {
 		"azurerm_security_center_subscription_pricing":                            resourceSecurityCenterSubscriptionPricing(),
 		"azurerm_security_center_workspace":                                       resourceSecurityCenterWorkspace(),
 		"azurerm_security_center_automation":                                      resourceSecurityCenterAutomation(),
-		"azurerm_security_center_auto_provisioning":                               resourceSecurityCenterAutoProvisioning(),
 		"azurerm_security_center_server_vulnerability_assessments_setting":        resourceSecurityCenterServerVulnerabilityAssessmentsSetting(),
 		"azurerm_security_center_server_vulnerability_assessment_virtual_machine": resourceServerVulnerabilityAssessmentVirtualMachine(),
+	}
+
+	if !features.FivePointOhBeta() {
+		resources["azurerm_security_center_auto_provisioning"] = resourceSecurityCenterAutoProvisioning()
 	}
 
 	return resources

--- a/internal/services/securitycenter/security_center_auto_provisioning_resource.go
+++ b/internal/services/securitycenter/security_center_auto_provisioning_resource.go
@@ -42,6 +42,8 @@ func resourceSecurityCenterAutoProvisioning() *pluginsdk.Resource {
 			Delete: pluginsdk.DefaultTimeout(60 * time.Minute),
 		},
 
+		DeprecationMessage: "The `azurerm_security_center_auto_provisioning` resource has been deprecated and will be removed in v5.0 of the AzureRM Provider",
+
 		Schema: map[string]*pluginsdk.Schema{
 			"auto_provision": {
 				Type:     pluginsdk.TypeString,

--- a/internal/services/securitycenter/security_center_auto_provisioning_resource_test.go
+++ b/internal/services/securitycenter/security_center_auto_provisioning_resource_test.go
@@ -19,6 +19,7 @@ import (
 type SecurityCenterAutoProvisionResource struct{}
 
 func TestAccSecurityCenterAutoProvision_update(t *testing.T) {
+	t.Skipf("Skipping since `azurerm_security_center_auto_provisioning` is deprecated on service side, can no longer be enabled.")
 	data := acceptance.BuildTestData(t, "azurerm_security_center_auto_provisioning", "test")
 	r := SecurityCenterAutoProvisionResource{}
 

--- a/website/docs/5.0-upgrade-guide.html.markdown
+++ b/website/docs/5.0-upgrade-guide.html.markdown
@@ -36,7 +36,7 @@ This deprecated resource has been superseded/retired and has been removed from t
 
 ### `azurerm_security_center_auto_provisioning`
 
-* This deprecated resource has been removed from the Azure Provider. [Please see the [documention for more details]](https://learn.microsoft.com/en-us/azure/defender-for-cloud/prepare-deprecation-log-analytics-mma-agent#log-analytics-agent-autoprovisioning-experience---deprecation-plan).
+* This deprecated resource has been removed from the Azure Provider. Please see the [documention for more details](https://learn.microsoft.com/en-us/azure/defender-for-cloud/prepare-deprecation-log-analytics-mma-agent#log-analytics-agent-autoprovisioning-experience---deprecation-plan).
 
 ## Removed Data Sources
 

--- a/website/docs/5.0-upgrade-guide.html.markdown
+++ b/website/docs/5.0-upgrade-guide.html.markdown
@@ -36,7 +36,7 @@ This deprecated resource has been superseded/retired and has been removed from t
 
 ### `azurerm_security_center_auto_provisioning`
 
-* This deprecated resource has been removed from the Azure Provider. [document](https://learn.microsoft.com/en-us/azure/defender-for-cloud/prepare-deprecation-log-analytics-mma-agent#log-analytics-agent-autoprovisioning-experience---deprecation-plan)
+* This deprecated resource has been removed from the Azure Provider. [Please see the [documention for more details]](https://learn.microsoft.com/en-us/azure/defender-for-cloud/prepare-deprecation-log-analytics-mma-agent#log-analytics-agent-autoprovisioning-experience---deprecation-plan).
 
 ## Removed Data Sources
 

--- a/website/docs/5.0-upgrade-guide.html.markdown
+++ b/website/docs/5.0-upgrade-guide.html.markdown
@@ -57,6 +57,10 @@ Please follow the format in the example below for listing breaking changes in re
 * The `example_property_with_changed_default` property now defaults to `NewDefault`.
 ```
 
+### `azurerm_security_center_auto_provisioning`
+
+* This deprecated resources has been removed from the Azure Provider.
+
 ### `azurerm_sentinel_alert_rule_fusion`
 
 * The deprecated `name` property has been removed.

--- a/website/docs/5.0-upgrade-guide.html.markdown
+++ b/website/docs/5.0-upgrade-guide.html.markdown
@@ -34,6 +34,9 @@ Please follow the format in the example below for adding removed resources:
 This deprecated resource has been superseded/retired and has been removed from the Azure Provider.
 ```
 
+### `azurerm_security_center_auto_provisioning`
+
+* This deprecated resource has been removed from the Azure Provider. [document](https://learn.microsoft.com/en-us/azure/defender-for-cloud/prepare-deprecation-log-analytics-mma-agent#log-analytics-agent-autoprovisioning-experience---deprecation-plan)
 
 ## Removed Data Sources
 
@@ -56,10 +59,6 @@ Please follow the format in the example below for listing breaking changes in re
 * The deprecated `example_property_with_no_replacement` property has been removed.
 * The `example_property_with_changed_default` property now defaults to `NewDefault`.
 ```
-
-### `azurerm_security_center_auto_provisioning`
-
-* This deprecated resources has been removed from the Azure Provider.
 
 ### `azurerm_sentinel_alert_rule_fusion`
 

--- a/website/docs/r/security_center_auto_provisioning.html.markdown
+++ b/website/docs/r/security_center_auto_provisioning.html.markdown
@@ -10,6 +10,8 @@ description: |-
 
 Enables or disables the Security Center Auto Provisioning feature for the subscription
 
+~> **Note:** The `azurerm_security_center_auto_provisioning` resource has been deprecated because [the auto provisioning capability will be deprecated by end of Novemember of 2024](https://learn.microsoft.com/en-us/azure/defender-for-cloud/prepare-deprecation-log-analytics-mma-agent#log-analytics-agent-autoprovisioning-experience---deprecation-plan) and will be removed in v5.0 of the AzureRM Provider.
+
 ~> **NOTE:** There is no resource name required, it will always be "default"
 
 ## Example Usage

